tests/cpp/dlr_relayvm_test.cc

```diff
@@ -0,0 +1,105 @@
+#include "dlr_relayvm.h"
+
+#include <gtest/gtest.h>
+#include "test_utils.hpp"
+
+int main(int argc, char **argv) {
+  testing::InitGoogleTest(&argc, argv);
+#ifndef _WIN32
+  testing::FLAGS_gtest_death_test_style = "threadsafe";
+#endif  // _WIN32
+  return RUN_ALL_TESTS();
+}
+
+class RelayVMTest : public ::testing::Test {
+ protected:
+  int8_t *img;
+  const int batch_size = 1;
+  size_t img_size = 512 * 512 * 3;
+  const int64_t input_shape[4] = {1, 512, 512, 3};
+  const int input_dim = 4;
+
+  dlr::RelayVMModel *model;
+
+  RelayVMTest() {
+    const int device_type = kDLCPU;
+    const int device_id = 0;
+    DLContext ctx = {static_cast<DLDeviceType>(device_type), device_id};
+    std::vector<std::string> paths = {"./ssd_mobilenet_v2"};
+    model = new dlr::RelayVMModel(paths, ctx);
+
+    img = new int8_t[img_size];
+  }
+
+  ~RelayVMTest() {
+    delete model;
+    delete img;
+  }
+};
+
+TEST_F(RelayVMTest, TestGetNumInputs) { EXPECT_EQ(model->GetNumInputs(), 1); }
+
+TEST_F(RelayVMTest, TestGetInputName) {
+  EXPECT_STREQ(model->GetInputName(0), "image_tensor");
+}
+
+TEST_F(RelayVMTest, TestGetInputType) {
+  EXPECT_STREQ(model->GetInputType(0), "uint8");
+}
+
+TEST_F(RelayVMTest, TestSetInput) {
+  EXPECT_NO_THROW(model->SetInput("image_tensor", input_shape, img, input_dim));
+}
+
+TEST_F(RelayVMTest, TestGetNumOutputs) { EXPECT_EQ(model->GetNumOutputs(), 3); }
+
+TEST_F(RelayVMTest, TestGetOutputName) {
+  EXPECT_STREQ(model->GetOutputName(0), "detection_scores:0");
+  EXPECT_STREQ(model->GetOutputName(1), "detection_classes:0");
+  EXPECT_STREQ(model->GetOutputName(2), "num_detections:0");
+}
+
+TEST_F(RelayVMTest, TestGetOutputType) {
+  EXPECT_STREQ(model->GetOutputType(0), "float32");
+  EXPECT_STREQ(model->GetOutputType(1), "float32");
+  EXPECT_STREQ(model->GetOutputType(2), "float32");
+}
+
+TEST_F(RelayVMTest, TestRun) {
+  EXPECT_NO_THROW(model->SetInput("image_tensor", input_shape, img, input_dim));
+  model->Run();
```

Comment:
As a unit test, I would suggest we create something simple and assert the result. Likely, you can find some example under tvm/tests/cpp/ for build_module.

Reply:
In the following tests we assert the shapes of the outputs.

Reply:
I don't see any tests that in the directory for vm. Could you point me to it.

Reply:
https://github.com/neo-ai/tvm/tree/dev/tests/cpp

Reply:
we should not only assert shapes but also check the returned values. All tests for vm are from python. So no gtest is available. We have gtest for build module https://github.com/apache/incubator-tvm/blob/master/tests/cpp/build_module_test.cc

Reply:
okay, let's skip this for now. We need to have a pre-compiled model for testing